interface/main/backup.php

```diff
@@ -136,12 +163,277 @@ function gzopen($filename, $mode, $use_include_path = 0)
+        echo '"' . xl('Label') . '",';
+        echo '"' . xl('Translated') . '",';
+        echo '"' . xl('UOR') . '",';
+        echo '"' . xl('Type') . '",';
+        echo '"' . xl('Width') . '",';
+        echo '"' . xl('Height') . '",';
+        echo '"' . xl('Max') . '",';
+        echo '"' . xl('List') . '",';
+        echo '"' . xl('Label Cols') . '",';
+        echo '"' . xl('Data Cols') . '",';
+        echo '"' . xl('Options') . '",';
+        echo '"' . xl('Description') . '",';
+        echo '"' . xl('Translated') . '",';
+        echo '"' . xl('Conditions') . '"';
+        echo "\n";
+        foreach ($_POST['form_sel_layouts'] as $layoutid) {
+            $res = sqlStatement(
+                "SELECT l.*, p.grp_title FROM layout_options AS l " .
+                "JOIN layout_group_properties AS p ON p.grp_form_id = l.form_id AND " .
+                "p.grp_group_id = l.group_id AND p.grp_activity = 1 " .
+                "WHERE l.form_id = ? ORDER BY l.group_id, l.seq, l.title",
+                array($layoutid)
+            );
+            while ($row = sqlFetchArray($res)) {
+                $xtitle = xl_layout_label($row['title']);
+                if ($xtitle === $row['title']) {
+                    $xtitle = '';
+                }
+                $xdesc = $row['description'];
+                if (substr($xdesc, 0, 1) != '<') {
+                    $xdesc = xl_layout_label($xdesc);
+                }
+                if ($xdesc === $row['description']) {
+                    $xdesc = '';
+                }
+                echo '"' . csvtext($row['form_id'     ]) . '",';
+                echo '"' . csvtext($row['seq'         ]) . '",';
+                echo '"' . csvtext($sources[$row['source']]) . '",';
+                echo '"' . csvtext($row['grp_title'   ]) . '",';
+                echo '"' . csvtext($row['field_id'    ]) . '",';
+                echo '"' . csvtext($row['title'       ]) . '",';
+                echo '"' . csvtext($xtitle) . '",';
+                echo '"' . csvtext($UOR[$row['uor']]) . '",';
+                echo '"' . csvtext($datatypes[$row['data_type']]) . '",';
+                echo '"' . csvtext($row['fld_length'  ]) . '",';
+                echo '"' . csvtext($row['fld_rows'    ]) . '",';
+                echo '"' . csvtext($row['max_length'  ]) . '",';
+                echo '"' . csvtext($row['list_id'     ]) . '",';
+                echo '"' . csvtext($row['titlecols'   ]) . '",';
+                echo '"' . csvtext($row['datacols'    ]) . '",';
+                echo '"' . csvtext($row['edit_options']) . '",';
+                echo '"' . csvtext($row['description' ]) . '",';
+                echo '"' . csvtext($xdesc) . '",';
+                echo '"' . csvtext($row['conditions'  ]) . '"';
+                echo "\n";
+            }
+        }
+    }
+    exit(0);
+}
+
+// CSV export of old log entries.
+//
+if ($form_step == 402) {
+    $end_date = fixDate($_POST['form_end_date'], '');
```

Comment:
to support internationalization, use the DateToYYYYMMDD() function here instead of fixDate